EmergencyStickerEncryption.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  EmergencyStickerEncryption.swift
+//  Krypt
+//
+//  Created by Sun Bin Kim on 21.06.19.
+//
+
+import CryptoSwift
+import Foundation
+
+public class EmergencyStickerEncryption {
+  /// Errors
+  ///
+  /// - invalidSalt: Failed to process salt
+  /// - failToGenerateKeyAndFingerprintFile: Failed to generate key and fingerprint file
+  public enum Error: LocalizedError {
+    case invalidSalt
+    case failToGenerateKeyAndFingerprintFile
+  }
+
+  /// Scrypt constants
+  private static let scryptR: Int = 10
+  private static let scryptN: Int = 16384
+  private static let scryptP: Int = 1
+  private static let scryptLength: Int = 64
+
+  /// Key and fingerprint file length for spliting hash
+  private static let fingerprintSecretLength: Int = 32
+  private static let keyLength: Int = 32
+  private static let fingerprintFileLength: Int = 32
+
+  /// AES block mode
+  private static let aesBlockMode: AES256.BlockMode = .gcm
+
+  /// First salt used during fingerprint secret generation
+  private static let firstSalt: String = "5f1288159017d636c13c1c1b2835b8a871780bc2"
+
+  private static let version: MedStickerEncryption.Version = .charlie
+
+  /// Holds key and fingerprint file
+  //  fingerprintFile is hex encoded and contains version in front
+  public struct KeyFingerprintFilePair {
+    public let key: Data
```

Comment:
```suggestion
    public let key: Key
```
Krypt is dealing only with `Krypt.Key`

Reply:
Isn't Key for RSA asym encryption?
This is AES sym encryption

Reply:
Yeah true, was too fast on that one 🙃